sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -125,6 +125,30 @@ gnrc_netif_t *gnrc_netif_iter(const gnrc_netif_t *prev)
     return (gnrc_netif_t*) netif_iter((netif_t*) prev);
 }
 
+gnrc_netif_t *gnrc_netif_get_by_type(netdev_type_t type, uint8_t index)
+{
+    gnrc_netif_t *netif = NULL;
+    while ((netif = gnrc_netif_iter(netif))) {
+
+#ifdef MODULE_NETDEV_REGISTER
+        if (netif->dev->type != type && type != NETDEV_ANY) {
+            continue;
+        }
+
+        if (netif->dev->index != index && index != NETDEV_INDEX_ANY) {
+            continue;
+        }
+#else
+        (void) type;
+        (void) index;
```

Comment:
Does it make sense to implement this function without `netdev_register` being used? IMO it should at least return `NULL` in that case, rather than just whatever interface by chance turns out to be first returned by `gnrc_netif_iter()`.

Reply:
The idea is that `netdev_register` is always used, unless `gnrc_netif_single` is used.

Reply:
I see.

Maybe add an `assert()` on `index` to be `0` if `gnrc_netif_single` is used?